tests/phpunit/api/v3/OptionValueCHTest.php

```diff
@@ -192,4 +192,99 @@ public function testCreateOptionValueChange() {
     $this->assertEmpty($updatedMap['values']);
   }
 
+  public function testContributionOnOptionvalueCHValueChange() {
+    $chFund = $this->callAPISuccess('OptionValue', 'create', [
+      'label' => 'Test Created CH Fund 1',
+      'option_group_id' => 'ch_fund',
+      'value' => 'CH+99999',
+    ]);
+
+    $contributionID = $this->callAPISuccess('CHContribution', 'create', [
+      'receive_date' => date('Ymd'),
+      'total_amount' => 100.00,
+      'payment_instrument_id' => 1,
+      'source' => 'SSF',
+      'contribution_status_id' => 1,
+      'ch_fund' => 'CH+99999'
+    ])['id'];
+    $contribution = $this->callAPISuccess('Contribution', 'getsingle', [
+      'id' => $contributionID,
+    ]);
+    // ensure that contribution Fund is assigned to 'Unassigned CH Fund'
+    $this->assertEquals($this->unallocatedFund['id'], $contribution['financial_type_id']);
+
+    // change the Fund of contribution to test that changing a CH fund option value doesn't impact the associated contribution's fund
```

Comment:
@monishdeb this feels very odd / wrong in that it seems to me that we have just set the financial type which doesn't match the ch_fund mapping because ch_fund at this point would be mapped to the unallocatedFund right?